lib/tasks/inactive_users.rake

```diff
@@ -0,0 +1,78 @@
+namespace :inactive_users do
+  desc 'List inactive users'
+  task list: :environment do
+    CSV.open(Rails.root.join('tmp/inactive_users.csv'), 'wb') do |csv|
+      csv << %w[
+        id
+        email
+        last_sign_in_at
+        current_sign_in_at
+        created_at
+        updated_at
+        persona_type
+      ]
+      inactive_users = User.where('disabled_at IS NULL AND (last_sign_in_at < :limit OR (last_sign_in_at IS NULL AND created_at < :limit))', limit: 6.months.ago.to_date)
+      inactive_users.each do |inactive_user|
+        csv << [
+          inactive_user.id,
+          inactive_user.email,
+          inactive_user.last_sign_in_at,
+          inactive_user.current_sign_in_at,
+          inactive_user.created_at,
+          inactive_user.updated_at,
+          inactive_user.persona_type
+        ]
+      end
+      puts "Listed #{inactive_users.count} inactive users in tmp/inactive_users.csv"
+    end
+  end
+
+  desc 'Disable inactive users'
+  task disable: :environment do
+    inactive_users = User.where('disabled_at IS NULL AND (last_sign_in_at < :limit OR (last_sign_in_at IS NULL AND created_at < :limit))', limit: 6.months.ago.to_date)
```

Comment:
As above,

```suggestion
    inactive_users = User.where(last_sign_in_at: ..6.months.ago).or(User.where(last_sign_in_at: nil, created_at: ..6.months.ago)).where(disabled_at: nil)
```

Also, can this go into a helper in (e.g.) `lib/tasks/rake_helpers/user_management.rb`?